.github/workflows/main.yaml

```diff
@@ -0,0 +1,40 @@
+name: main
+on: [push, pull_request]
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniconda-version: "latest"
+          mamba-version: "*"
+          environment-file: conda/local.yaml
+          channels: conda-forge,nodefaults
+          activate-environment: waveapp
+          use-mamba: true
+          miniforge-variant: Mambaforge
+
+      - name: Install dependencies
+        run: |
+          curl -sSL https://install.python-poetry.org/ | python -
+          export PATH="$HOME/.poetry/bin:$PATH"
+
+      - name: install from poetry
+        run: poetry config virtualenvs.create true && poetry install
+
+      - name: check dir
+        run: |
+          pwd
+          ls -la
+
+      - name: run wave app
+        run: |
+          wave run app/app.py &
+          sleep 5 &
```

Comment:
maybe a test with wget or curl .. would be much better than just sleep

Reply:
Something like that?

```
      - name: run wave app
        run: |
          wave run app/app.py &
          curl -X GET http://127.0.0.1:10101 &
          kill %%
 ```

Reply:
I found this:
```
CODE=`curl --write-out '%{http_code}' \
    --silent \
    --output /dev/null \
    --request PUT \
    --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
    --header 'content-type: application/json' \
    --url 'https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.inputs.pr_num }}/merge' \
    --data '{ \
    "merge_method": "squash", \
    "commit_title": "Squash and Merge for Pull Request ${{ github.event.inputs.pr_num }}", \
    "commit_message": "expected sha value = ${{ github.event.inputs.pr_head_sha }}", \
    "sha":"${{ github.event.inputs.pr_head_sha }}" \
    }'`

if [ $CODE!="200" ] 
then
    echo "FAILURE"
else
    echo "SUCCESS"
fi
```

Reply:
Why is this necessary? Is this to verify that CI has successfully started the app?

Reply:
@fccoelho, yes.

Reply:
@esloch, mayme you can add a timeout there, something like:

`timeout 30 curl -X GET http://127.0.0.1:10101`

Reply:
@xmnlab, I tried two forms; one using timeout for curl as suggested, and another using timeout for waveapp, in both cases the step fails.

Reply:
> timeout 30 wave run app/app.py

actually it doesn't make sense to have timeout there .. because it should be running on background in order to test the connection with the service. also, this is a service that will be running in loop ... so if you add the timeout it will break the service.

Reply:
ok, but it looks like curls is being fired even before it raises the app:

```
      - name: Run wave app
        run: |
          wave run app/app.py &
          timeout 30 curl -X GET http://127.0.0.1:10101
```

```
Run wave run app/app.py &
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (7) Failed to connect to 127.0.0.1 port [10](https://github.com/thegraphnetwork/wave-docker-template/actions/runs/3788012643/jobs/6440362434#step:6:11)101 after 1 ms: Connection refused
``

Reply:
Yep that is right, probably you will need to have a script that checks the url in a loop and check until the status is 200 ( or any other code if correspond)

Reply:
@xmnlab. It worked, what do you think?

*Makefile:*
```
#  APP ON CI
.PHONY: app-wait
app-wait:
	timeout ${TIMEOUT} ./scripts/ci/status_code_check.sh
``` 
*status_code_check.sh:*
```
#!/usr/bin/env bash

echo "[I] Checking response status code from App..."

while [ "`curl -i http://localhost:10101 2>/dev/null | head -n 1 | cut -d$' ' -f2`" != 200 ];
do
    echo "[II] Waiting for Curl..."
    sleep 5;
done

echo "[III] The WaveApp is 200 OK!."
```